src/highdicom/ann/content.py

```diff
@@ -496,6 +498,73 @@ def primary_anatomic_structures(self) -> List[CodedConcept]:
             return []
         return list(self.PrimaryAnatomicStructureSequence)
 
+    def get_graphic_data(
+        self,
+        coordinate_type: Union[str, AnnotationCoordinateTypeValues]
+    ) -> List[np.ndarray]:
+        """Get spatial coordinates of all graphical annotations.
+
+        Parameters
+        ----------
+        coordinate_type: Union[str, highdicom.ann.AnnotationCoordinateTypeValues]
+            Coordinate type of annotation
+
+        Returns
+        -------
+        List[numpy.ndarray]
+            Two-dimensional array of floating-point values representing either
+            2D or 3D spatial coordinates for each graphical annotation
+
+        """  # noqa: E501
+        coordinate_type = AnnotationCoordinateTypeValues(coordinate_type)
+        if self._graphic_data is None:
+            if coordinate_type == AnnotationCoordinateTypeValues.SCOORD:
+                coordinate_dimensionality = 2
+            else:
+                coordinate_dimensionality = 3
+
+            try:
+                coordinates_data = getattr(self, 'DoublePointCoordinatesData')
+                coordinates_dtype = np.float64
+            except AttributeError:
+                coordinates_data = getattr(self, 'PointCoordinatesData')
+                coordinates_dtype = np.float32
+            decoded_coordinates_data = np.frombuffer(
+                coordinates_data,
+                coordinates_dtype
+            )
+
+            self._graphic_data = []
+            for i in range(self.number_of_annotations):
+                annotation_number = i + 1
+                coordinate_index = self._get_coordinate_index(
```

Comment:
Take it or leave it... but if I'm not mistaken the length of the annotations is fixed in the case of a RECTANGLE, POINT, or ELLIPSE. In these cases the entire loop here could be replaced with a basic array reshape, which would be very significantly faster. If you're worried about efficiency at all this is probably worth implementing two different cases (one for RECTANGLE, POINT, ELLIPSE and the other for POLYGON, POLYLINE).

Reply:
In fact, now that I think about it, I think you may be able to use `np.split()` and the point index list to replace the loop even in the polygon/polyline case. You'd just have to figure out what to do in the common z value case.

Reply:
Nice one! Addressed via 0b5096527929f83376167b518c9588b7b1b510f2.

We still need one loop (now implemented via a list comprehension) to reshape the coordinates and include the z-coordinate. But it should be faster.

Reply:
Nice! Addressed via 852dc1d94446db09d2b8025b4b6e14ca637f2197